### PR TITLE
feat(ARRR-239) - New Prop 'mode' for ["display" , "default"]

### PR DESCRIPTION
**Pull request by ibanzajoe**
### Description
ARRR-239
https://bambee.atlassian.net/jira/software/projects/ARRR/boards/63?assignee=610ad23b4e8d8d00699456e4&selectedIssue=ARRR-239

Need to add a prop to the component so that DEVELOPER can toggle the disabled state on/off for the survey.

### Screen Shots
mode: 'display':
image

mode: 'default':
image

### CODE REVIEW

File: src/BambeeSurvey.vue [hunk 1]
```diff
@@ -40,6 +44,10 @@ export default {
       this.$emit('complete', sender);
     });
 
+    if (this.mode === 'display') {
+      this.survey.mode = 'display';
```
Comment (uptownhr) [inline src/BambeeSurvey.vue:48]: quite confusing naming going on. @AlexCloz please chime in here and confirm what needs to happen. 

1. `this.survey` is actually questions.
2. we need to set `mode = 'display'` on `model`
3. model is Surveyjs instance?

Comment (AlexCloz) [reply src/BambeeSurvey.vue:48]: Correct. Should be `this.model.mode`.

File: src/BambeeSurvey.vue [hunk 2]
```diff
@@ -24,6 +24,10 @@ export default {
       type: Object,
       default: () => ({}),
     },
+    mode: {
+      type: String,
+      default: 'default',
```
Comment (AlexCloz) [inline src/BambeeSurvey.vue:29]: It would be beneficial to add validation on the `mode`

Comment (ibanzajoe) [reply src/BambeeSurvey.vue]: @AlexCloz Wrote a test in `BambeeSurvey.test.js` to validate that the `propData: mode` is correctly disabling the input fields

File: src/BambeeSurvey.vue (hunk 1 above)
Comment (ibanzajoe) [reply src/BambeeSurvey.vue]: changed `this.model.mode` from `this.survey.mode`.  Strange that `this.survey.mode = 'display'` still worked to disable the input fields.  

Also, why `display` is making input fields disabled?  wouldn't ['disabled', 'default'] be better?

**Review by uptownhr — APPROVED** [review]

**Review by AlexCloz — APPROVED** [review]

**Pull request merged**